dpnp/backend/examples/example_bs.cpp

```diff
@@ -0,0 +1,198 @@
+//*****************************************************************************
+// Copyright (c) 2016-2020, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+// - Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+// - Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+/**
+ * Example 3.
+ *
+ * This example shows simple usage of the DPNP C++ Backend library
+ * to calculate black scholes algo
+ *
+ * Possible compile line:
+ * clang++ -g -fPIC dpnp/backend/examples/example_bs.cpp -Idpnp -Idpnp/backend/include -Ldpnp -Wl,-rpath='$ORIGIN'/dpnp -ldpnp_backend_c -o example_bs
+ *
+ */
+
+#include <iostream>
+#include <cmath>
+
+#include "dpnp_iface.hpp"
+
+double* black_scholes_put(double* S, double* K, double* T, double* sigmas, double* r_sigma_sigma_2,
+                          double* nrs, double* sqrt2, double* ones, double* twos, const size_t size)
+{
+    double* d1 = (double*)dpnp_memory_alloc_c(size * sizeof(double));
+    dpnp_divide_c<double, double, double>(S, K, d1, size);         // S/K
+    dpnp_log_c<double, double>(d1, d1, size);               // np.log(S/K)
+
+    double* r_sigma_sigma_2_t = (double*)dpnp_memory_alloc_c(size * sizeof(double));
+    dpnp_multiply_c<double, double, double>(r_sigma_sigma_2, T, r_sigma_sigma_2_t, size);  // r_sigma_sigma_2*T
+    dpnp_add_c<double, double, double>(d1, r_sigma_sigma_2_t, d1, size);     // np.log(S/K) + r_sigma_sigma_2*T
+    dpnp_memory_free_c(r_sigma_sigma_2_t);
+
+    double* sigmas_sqrtt = (double*)dpnp_memory_alloc_c(size * sizeof(double));
```

Comment:
You are allocating mem on 49 and then deallocating it on 52 line, after you allocating the same memory on 54th line. Please do not make multiple allocations on your code, just reuse once allocated mem.